### PR TITLE
Rename CommandType to Name on resource command types

### PR DESCRIPTION
**Pull request by JamesNK**
## Description

Feedback from https://github.com/dotnet/aspire/pull/6295#discussion_r1803777740.

Note that this is a source breaking change for people who have implemented the proto when they update the proto the identifer will change name. It's not a wire breaking change because the field id (`1` in this case) is the same.

@joperezr Where should the `commandType` -> `name` breaking change be communicated? The model API is new in 9.0 so I believe can change at this point without breaking change notification.

## Checklist

- Is this feature complete?
  - [x] Yes. Ready to ship.
  - [ ] No. Follow-up changes expected.
- Are you including unit tests for the changes and scenario tests if relevant?
  - [ ] Yes
  - [x] No
- Did you add public API?
  - [x] Yes
    - If yes, did you have an API Review for it?
      - [ ] Yes
      - [ ] No
    - Did you add `<remarks />` and `<code />` elements on your triple slash comments?
      - [ ] Yes
      - [ ] No
  - [ ] No
- Does the change make any security assumptions or guarantees?
  - [ ] Yes
    - If yes, have you done a threat model and had a security review?
      - [ ] Yes
      - [ ] No
  - [x] No
- Does the change require an update in our Aspire docs?
  - [x] Yes
    - Link to aspire-docs issue: 
  - [ ] No

### CODE REVIEW

File: tests/Aspire.Hosting.Tests/Dashboard/DashboardServiceTests.cs [hunk 1]
```diff
@@ -79,7 +79,7 @@ public async Task WatchResources_ResourceHasCommands_CommandsSentWithResponse()
         using var applicationBuilder = TestDistributedApplicationBuilder.Create();
         var builder = applicationBuilder.AddResource(testResource);
         builder.WithCommand(
-            type: "TestType",
+            name: "TestType",
```
Comment (drewnoakes) [inline tests/Aspire.Hosting.Tests/Dashboard/DashboardServiceTests.cs:82]:
```suggestion
            name: "TestName",
```

**Review by drewnoakes — APPROVED** [review]
Looks fine, other than compile error in tests. I think "Id" would be a better term as it indicates the uniqueness requirement more so than "Name".

**Comment by DamianEdwards** [comment]
I think we need to do regular breaking change notifications as we changed this after the rc.

**Comment by JamesNK** [comment]
I prefer name because resources have a name, and endpoints have a name. Command names would be consistent.

And I think name makes more sense because the value is being provided by the user. Id feels more appropriate if the value is auto-generated.

**Review by davidfowl — APPROVED** [review]

**Comment by davidfowl** [comment]
It turns out we shipped commands in the resource server protocol so this change breaks ACA and we need to either revert all of it, or deprecate the old fields and add the new fields

**Pull request merged**